### ISSUE TITLE
lifelines doesn't support pandas' timedelta

### ISSUE THREAD

**Issue by HarvsG** [CamDavidsonPilon/lifelines#621]
I am trying to generate a Kaplan Meier curve from data `df['date_of_death_or_censor']` and `df['enroll_date']` to generate a time interval to death or censor. And then `df['Dead']` a boolean series describing whether death occurred `True` or censor occurred `False`

```
from lifelines import KaplanMeierFitter
kmf = KaplanMeierFitter()
T = df['date_of_death_or_censor'] - df['enroll_date']
E = df['Dead']
kmf.fit(T, event_observed=E)
```

where `T` is dtype `timedelta64[ns]`
Generates output:

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-23-882a3a6617eb> in <module>
      3 T = df['date_of_death_or_censor'] - echo['enroll_date']
      4 E = df['Dead']
----> 5 kmf.fit(T, event_observed=E)

~/.venv/jns/lib/python3.5/site-packages/lifelines/fitters/kaplan_meier_fitter.py in fit(self, durations, event_observed, timeline, entry, label, alpha, left_censorship, ci_labels, weights)
     95         # if the user is interested in left-censorship, we return the cumulative_density_, no survival_function_,
     96         estimate_name = "survival_function_" if not left_censorship else "cumulative_density_"
---> 97         v = _preprocess_inputs(durations, event_observed, timeline, entry, weights)
     98         self.durations, self.event_observed, self.timeline, self.entry, self.event_table = v
     99 

~/.venv/jns/lib/python3.5/site-packages/lifelines/utils/__init__.py in _preprocess_inputs(durations, event_observed, timeline, entry, weights)
    801         entry = np.asarray(entry).reshape((n,))
    802 
--> 803     event_table = survival_table_from_events(durations, event_observed, entry, weights=weights)
    804     if timeline is None:
    805         timeline = event_table.index.values

~/.venv/jns/lib/python3.5/site-packages/lifelines/utils/__init__.py in survival_table_from_events(death_times, event_observed, birth_times, columns, weights, collapse, intervals)
    331     death_times = np.asarray(death_times)
    332     if birth_times is None:
--> 333         birth_times = min(0, death_times.min()) * np.ones(death_times.shape[0])
    334     else:
    335         birth_times = np.asarray(birth_times)

TypeError: Cannot cast ufunc less input from dtype('<m8[ns]') to dtype('<m8') with casting rule 'same_kind'
```

**Comment by HarvsG** [comment]
a quick fix is to do:

```
from lifelines import KaplanMeierFitter
kmf = KaplanMeierFitter()
T = df['date_of_death_or_censor'] - df['enroll_date']

T = T.dt.days

E = df['Dead']
kmf.fit(T, event_observed=E)
```

**Comment by CamDavidsonPilon** [comment]
Hello! I'll try to fix this for 0.18.0 release. Thanks for the report.

**Comment by CamDavidsonPilon** [comment]
^ that being said, I'll add a check to make sure users aren't giving me timedeltas, and print our a better error message.
